File: px-services/src/main/java/com/mercadopago/android/px/internal/core/PlatformInterceptor.java
package com.mercadopago.android.px.internal.core;

import android.content.Context;
import android.support.annotation.NonNull;
import java.io.IOException;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public final class PlatformInterceptor implements Interceptor {

    private static final String HEADER_PLATFORM = "x-platform";
    private static final String PLATFORM_MP = "MP";
    private static final String PLATFORM_ML = "ML";
    private Context context;

    public PlatformInterceptor(@NonNull final Context context) {
        this.context = context.getApplicationContext();
    }

    @Override
    public Response intercept(final Chain chain) throws IOException {
        final Request originalRequest = chain.request();
        final Request request = originalRequest.newBuilder()
            .header(HEADER_PLATFORM, getPlatform(context))
            .build();
        return chain.proceed(request);
    }


    private String getPlatform(final Context context) {
        final String packageName = context.getApplicationInfo().packageName;
        return packageName.contains("com.mercadolibre") ? PLATFORM_ML : PLATFORM_MP;
    }
}

File: px-services/src/main/java/com/mercadopago/android/px/internal/core/PlatformInterceptor.kt
package com.mercadopago.android.px.internal.core

import android.content.Context
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class PlatformInterceptor(context: Context) : Interceptor {
    private val currentPlatform = getPlatform(context)

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalRequest = chain.request()
        val request = originalRequest.newBuilder()
            .header(HEADER_PLATFORM, currentPlatform)
            .build()
        return chain.proceed(request)
    }

    private fun getPlatform(context: Context): String {
        val packageName = context.applicationInfo.packageName
        return if (packageName.contains("com.mercadolibre")) PLATFORM_ML else PLATFORM_MP
    }

    companion object {
        private const val HEADER_PLATFORM = "x-platform"
        private const val PLATFORM_MP = "MP"
        private const val PLATFORM_ML = "ML"
    }
}